src/cmd/cmd_conc.ml

```diff
@@ -419,19 +419,45 @@ let run solver tree link_state modules_to_run =
   in
   loop initial_model
 
+let assignments_to_model (assignments : (Smtml.Symbol.t * V.t) list) :
+  Smtml.Model.t =
+  let table = Hashtbl.create (List.length assignments) in
+  List.iter
+    (fun (s, v) ->
+      let value =
+        match v with
+        | Concrete_value.I32 x -> Smtml.Value.Num (I32 x)
+        | I64 x -> Num (I64 x)
+        | F32 x -> Num (F32 (Float32.to_bits x))
+        | F64 x -> Num (F64 (Float64.to_bits x))
+        | Ref _ -> assert false
+      in
+      Hashtbl.add table s value )
+    assignments;
+  table
+
 (* NB: This function propagates potential errors (Result.err) occurring
    during evaluation (OS, syntax error, etc.), except for Trap and Assert,
    which are handled here. Most of the computations are done in the Result
    monad, hence the let*. *)
 let cmd ~profiling ~debug ~unsafe ~rac ~srac ~optimize ~workers:_
   ~no_stop_at_failure:_ ~no_value ~no_assert_failure_expression_printing
   ~deterministic_result_order:_ ~fail_mode:_ ~(workspace : Fpath.t) ~solver
-  ~files ~profile =
+  ~files ~profile ~model_output_format =
   Option.iter Stats.init_logger_to_file profile;
   if profiling then Log.profiling_on := true;
   if debug then Log.debug_on := true;
   (* deterministic_result_order implies no_stop_at_failure *)
   (* let no_stop_at_failure = deterministic_result_order || no_stop_at_failure in *)
+  let to_string =
+    match String.lowercase_ascii model_output_format with
+    | "json" -> fun m -> assignments_to_model m |> Smtml.Model.to_json_string
+    | "scfg" ->
+      fun m -> assignments_to_model m |> Smtml.Model.to_scfg_string ~no_value
+    | _ ->
+      Fmt.epr "Expected \"json\" or \"scfg\" but got %s\n" model_output_format;
+      assert false
+  in
```

Comment:
It would be better to put this in `bin/owi.ml` and write a cmdliner `model_format_conv` with a dedicated type (`type model_output_format = Scfg | Json`) so that it is not duplicated here and in symbolic.